makepkg.py

```diff
@@ -48,62 +49,77 @@
-
-def do(*args):  # pylint: disable=C0103,W0621
+def do(*args):
     try:
-        subprocess.check_output(args, stderr=subprocess.STDOUT)
+        return subprocess.check_output(args, stderr=subprocess.STDOUT)
     except subprocess.CalledProcessError as e:  # pylint: disable=C0103
-        print("ERROR: {0}".format(str(e)))
-        print("\t{0}".format(" ".join(args)))
-        print(e.output)
-        sys.exit(1)
-
+        raise Exception("[{0}] failed:\n{1}\n{2}".format(" ".join(args), str(e), e.stdout))
+
+
+def run(agent_family, output_directory, log):
+    output_path = os.path.join(output_directory, "eggs")
+    target_path = os.path.join(output_path, AGENT_LONG_VERSION)
+    bin_path = os.path.join(target_path, "bin")
+    egg_path = os.path.join(bin_path, AGENT_LONG_VERSION + ".egg")
+    manifest_path = os.path.join(target_path, AGENT_MANIFEST_FILE)
+    publish_manifest_path = os.path.join(target_path, PUBLISH_MANIFEST_FILE)
+    pkg_name = os.path.join(output_path, AGENT_LONG_VERSION + ".zip")
+
+    if os.path.isdir(target_path):
+        shutil.rmtree(target_path)
+    elif os.path.isfile(target_path):
+        os.remove(target_path)
+    if os.path.isfile(pkg_name):
+        os.remove(pkg_name)
+    os.makedirs(bin_path)
+    log.info("Created {0} directory".format(target_path))
+
+    setup_script = str(pathlib.Path(__file__).parent.joinpath("setup.py"))
+    args = ["python3", setup_script, "bdist_egg", "--dist-dir={0}".format(bin_path)]
+
+    log.info("Creating egg {0}".format(egg_path))
+    do(*args)
+
+    egg_name = os.path.join("bin", os.path.basename(
+        glob.glob(os.path.join(bin_path, "*"))[0]))
+
+    log.info("Writing {0}".format(manifest_path))
+    with open(manifest_path, mode='w') as manifest:
+        manifest.write(MANIFEST.format(AGENT_NAME, egg_name))
+
+    log.info("Writing {0}".format(publish_manifest_path))
+    with open(publish_manifest_path, mode='w') as publish_manifest:
+        publish_manifest.write(PUBLISH_MANIFEST.format(AGENT_VERSION, agent_family))
+
+    cwd = os.getcwd()
+    os.chdir(target_path)
+    try:
+        log.info("Creating package {0}".format(pkg_name))
+        do("zip", "-r", pkg_name, egg_name)
+        do("zip", "-j", pkg_name, AGENT_MANIFEST_FILE)
+        do("zip", "-j", pkg_name, PUBLISH_MANIFEST_FILE)
+    finally:
+        os.chdir(cwd)
 
-if os.path.isdir(target_path):
-    shutil.rmtree(target_path)
-elif os.path.isfile(target_path):
-    os.remove(target_path)
-if os.path.isfile(pkg_name):
-    os.remove(pkg_name)
-os.makedirs(bin_path)
-print("Created {0} directory".format(target_path))
+    log.info("Package {0} successfully created".format(pkg_name))
 
-args = ["python", "setup.py", "bdist_egg", "--dist-dir={0}".format(bin_path)]  # pylint: disable=invalid-name
 
-print("Creating egg {0}".format(egg_path))
-do(*args)
+if __name__ == "__main__":
+    logging.basicConfig(format='%(message)s', level=logging.INFO)
```

Comment:
where are the log msgs stored? which file?

Reply:
When the py file is invoked as a script (__name__ == "__main__") we just  to output to stdout/stderr , no file is created. Same behavior as the previous version of this script.